avro2tf/src/main/scala/com/linkedin/avro2tf/jobs/FeatureListGeneration.scala

```diff
@@ -104,32 +121,45 @@ class FeatureListGeneration {
     dataFrame: DataFrame,
     params: TensorizeInParams,
     fileSystem: FileSystem,
-    colsToCollectFeatureList: Seq[String]): Unit = {
+    colsToCollectFeatureList: Seq[String]): mutable.HashSet[String] = {
 
     import dataFrame.sparkSession.implicits._
     val dataFrameSchema = dataFrame.schema
     val tmpFeatureListPath = s"${params.workingDir.rootPath}/$TMP_FEATURE_LIST"
     fileSystem.delete(new Path(tmpFeatureListPath), ENABLE_RECURSIVE)
     val outputTensorDataTypes = TensorizeInConfigHelper.getOutputTensorDataTypes(params)
 
+    // collect NTV tensors
```

Comment:
Should move this logic outside this method, and possibly in a new method.

Reply:
Good point. Will address.